test: fixed failing missed k8s tests

fristonio opened the pull request:
See individual commit messages for more details.

Fixes #12867

fristonio committed:
helm: change serviceaccount name for cilium-pre-flight components

* This commit changes service account name for cilium-pre-flight check
  so that it does not conflict with cilium serviceAccount name.

Signed-off-by: Deepesh Pathak <deepshpathak@gmail.com>

fristonio committed:
test/k8sT: do proper cleanup in update tests

* Do proper cleanup of cilium deployment in K8s update test
* Fix minor issue with DeploymentManager DeleteCilium function.
* Add logging for context in UninstallCiliumFromManifest

Signed-off-by: Deepesh Pathak <deepshpathak@gmail.com>

fristonio committed:
test/k8sT: fix failing missed k8s tests

* Remove checking the presence of manifest file using os.Stat this does
  not work for some reason when running missed k8s tests. Instead create
  a dummy file if it does not exist and run `kubectl delete` on that so
  if the file does not exist it won't throw the kubectl error.
* Remove dead code where we fallback to deleting `cilium.yaml` instead of
  timestamped file that we now use everywhere.

Signed-off-by: Deepesh Pathak <deepshpathak@gmail.com>

aanm commented on install/kubernetes/cilium/charts/preflight/templates/clusterrolebinding.yaml:
```diff
@@ -8,5 +8,5 @@ roleRef:
 subjects:
 - kind: ServiceAccount
-  name: cilium
+  name: cilium-pre-flight
```
Please add in the commit message the "Why?" we need this change.

aanm reviewed (changes requested):
Small nit

christarazi reviewed (approved):
LGTM, there's a typo in one of serviceAccount commit: `serviceAccnout`

fristonio committed:
test/helpers: fix spelling of ciliumclusterwidenetworkpolicy crd

* Fixes the spelling of ciliumclusterwidenetworkpolicy rbac resource we dump
  when gathering log using kubectl helper.

Fixes: aeefd5862ae4 ("test/helpers: dump rbac information on k8s test failure")

Signed-off-by: Deepesh Pathak <deepshpathak@gmail.com>

sayboras reviewed (approved):
LGTM and thanks.

I can re-run gingko test with `--cilium.provision=false` now :man_dancing:

fristonio committed:
test/helpers: cleanup cilium components before running any tests

This commits adds a cleaning up step in BeforeAll block of test suite
that make sure to clean up any orphaned cilium components in the
cluster. This is useful when running local tests which in some cases
like improper cleanup might lead to orphaned cilium components.

Signed-off-by: Deepesh Pathak <deepshpathak@gmail.com>

aanm commented on test/helpers/kubectl.go:
```diff
@@ -3268,7 +3268,7 @@ func (kub *Kubectl) GatherLogs(ctx context.Context) {
 		"kubectl get replicationcontroller --all-namespaces -o json":         "replicationcontroller.txt",
```
In the commit message, it should be:
```
Fixes: aeefd5862ae4 ("test/helpers: dump rbac information on k8s test failure")
Signed-off-by: Deepesh Pathak <deepshpathak@gmail.com>
```
